Modules/Relationships/src/Relationships.Application/DomainEvents/Incoming/FeatureFlagsOfIdentityChanged/FeatureFlagsOfIdentityChangedDomainEventHandler.cs

```diff
@@ -37,31 +39,23 @@ private async Task<HashSet<IdentityAddress>> FindAllIdentitiesToBeNotified(Featu
     {
         var identitiesToBeNotified = new HashSet<IdentityAddress>();
 
-        var activeAndPendingRelationships = await
+        var activeAndPendingRelationshipAddressPairs = await
             _relationshipsRepository.FindRelationships(
-                r => (r.From == @event.IdentityAddress || r.To == @event.IdentityAddress)
-                     && (r.Status == RelationshipStatus.Active || r.Status == RelationshipStatus.Pending),
+                Relationship.HasParticipant(@event.IdentityAddress).And(Relationship.HasStatusInWhichPeerShouldBeNotifiedAboutDeletion()),
+                r => new { r.From, r.To },
                 CancellationToken.None);
 
-        foreach (var relationship in activeAndPendingRelationships)
-        {
-            if (relationship.From == @event.IdentityAddress)
-            {
-                identitiesToBeNotified.Add(relationship.To);
-            }
-            else
-            {
-                identitiesToBeNotified.Add(relationship.From);
-            }
-        }
+        foreach (var addressPair in activeAndPendingRelationshipAddressPairs)
+            identitiesToBeNotified.Add(addressPair.From == @event.IdentityAddress ? addressPair.To : addressPair.From);
```

Comment:
For me it's much easier to understand this if you turn around the operands of the `==`:

```suggestion
            identitiesToBeNotified.Add(@event.IdentityAddress == addressPair.From ? addressPair.To : addressPair.From);
```

If you think so too, you can turn apply my suggestion. But if you think different, feel free to keep it as it is.